src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resource/TemporaryAbsencesResourceTest.kt

```diff
@@ -53,7 +53,39 @@ class TemporaryAbsencesResourceTest : IntegrationTestBase() {
         .exchange()
         .expectStatus().isOk
 
-      verify(prisonService).getTemporaryAbsences("MDI")
+      verify(temporaryAbsenceService).getTemporaryAbsences("MDI")
+    }
+  }
+
+  @Nested
+  inner class `Get Temporary absence` {
+
+    @Test
+    fun `returns json in expected format`() {
+      whenever(temporaryAbsenceService.getTemporaryAbsence("MDI", "A1234AA")).thenReturn(
+        TemporaryAbsence(
+          firstName = "Jim",
+          lastName = "Smith",
+          dateOfBirth = LocalDate.of(1991, 7, 31),
+          prisonNumber = "A1234AA",
+          reasonForAbsence = "Hospital"
+        )
+      )
+      webTestClient.get().uri("/temporary-absences/MDI/A1234AA")
+        .headers(setAuthorisation(roles = listOf("ROLE_VIEW_ARRIVALS"), scopes = listOf("read")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody().json("temporaryAbsence".loadJson(this))
+    }
```

Comment:
Maybe best to add tests for?:
* `requires correct role`
* `requires authentication`

Reply:
Added in latest commit